style: apply /style-guide pass to models/artifacts (#2717)
wandb/docs: models/artifacts/delete-artifacts.mdx

title: Delete an artifact
---

Delete artifacts interactively with the W&B App or programmatically with the W&B Python SDK. When you delete an artifact, W&B marks that artifact as a *soft-delete*. In other words, W&B marks the artifact for deletion but doesn't immediately delete files from storage.

The contents of the artifact remain as a soft-delete, or pending deletion state, until a regularly run garbage collection process reviews all artifacts marked for deletion. The garbage collection process deletes associated files from storage if a previous or subsequent artifact version doesn't use the artifact and its associated files.

<Note>
Garbage collection is **best-effort**. W&B does not guarantee how quickly freed space appears in your object storage after you delete an artifact. Large deployments or backlogs can take longer than expected. For how this fits with run data, retention settings, and optional operator actions, see [Manage bucket storage and costs](/platform/hosting/managing-bucket-storage).
</Note>

## Artifact garbage collection workflow

> This section provides a visual overview of how artifact deletion flows through soft-delete and garbage collection before files are permanently removed. The following diagram illustrates the complete artifact garbage collection process:

Reviewer: ```suggestion
The following diagram illustrates the complete artifact garbage collection process:
```

Author: Applied your lead-in; removed the "visual overview" filler sentence.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

    style KeepFiles fill:#e8f5e9,stroke:#333,stroke-width:2px,color:#000
    style DeleteFiles fill:#ffebee,stroke:#333,stroke-width:2px,color:#000
    style End fill:#e0e0e0,stroke:#333,stroke-width:2px,color:#000
```

You can schedule when artifacts are deleted from W&B with TTL policies. For more information, see [Manage data retention with Artifact TTL policy](./ttl).

<Note>

> Artifacts deleted by a TTL policy, the W&B Python SDK, or the W&B App are first soft-deleted. Soft-deleted artifacts are then garbage-collected before they're permanently deleted.

Reviewer: ```suggestion
Artifacts deleted by a TTL policy, the W&B Python SDK, or the W&B App are first soft-deleted. Soft-deleted artifacts are then garbage-collected before they are permanently deleted.
```

Author: Applied your wording ("…before they are permanently deleted").

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Delete multiple artifact versions

> Use this approach when you want to clean up several artifact versions logged by a single run in one operation, rather than deleting each version individually. The following code example shows how to delete multiple artifact versions. Provide the entity, project name, and run ID that created the artifact as arguments to `wandb.Api.run()`. This returns a run object that you can use to access all artifact versions created by that run. Next, iterate through the artifact versions and delete the versions that match your criteria.

Reviewer: * "...clean up.." is informal
* Which approach? 
* Not necessarily true, there can be numerous reasons + the example is just one way to do it. THere are other ways that depend on the story.
```suggestion
The following code example shows how to delete multiple artifact versions. Provide the entity, project name, and run ID that created the artifact as arguments to `wandb.Api.run()`. This returns a run object that you can use to access all artifact versions created by that run. Next, iterate through the artifact versions and delete the versions that match your criteria.
```

Author: Applied your version; removed the "clean up / single run" framing.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Delete multiple artifact versions with a specific alias

> To remove only versions tagged with a particular alias (for example, an obsolete release tag), filter on the alias before deleting. The following code demonstrates how to delete multiple artifact versions that have a specific alias.

Reviewer: "...filter on the alias..." what?

Author: Removed the unclear "filter on the alias" sentence.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Delete an artifact collection

> Deleting a collection removes the collection and all artifact versions it contains. Deleted versions follow the same soft-delete and garbage collection workflow described in the preceding sections.

Reviewer: Extraneous + In general dangerous: don't add any info about data usage that we haven't confirmed. 

```suggestion
Deleting a collection removes the collection and all artifact versions it contains.
```

Author: Applied — dropped the unconfirmed data-usage claim; now just "Deleting a collection removes the collection and all artifact versions it contains."

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

<Note>Garbage collection timing is not guaranteed. See [Manage bucket storage and costs](/platform/hosting/managing-bucket-storage) for details.</Note>

> Whether garbage collection is active by default depends on your W&B deployment type, so the steps to enable it vary. Garbage collection is active by default if you use W&B Multi-tenant Cloud. In W&B Dedicated and Self-Managed, you might need to take the following steps to activate garbage collection.

Reviewer: ```suggestion
Garbage collection behavior depends on your W&B deployment type. For W&B Multi-tenant Cloud, garbage collection is enabled by default. For W&B Dedicated and Self-Managed deployments, you might need to enable garbage collection manually.
```

Author: Applied your deployment-type wording.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
